koku/api/settings/tag_mappings/view.py

```diff
@@ -0,0 +1,176 @@
+        paginator = ListPaginator(serializer.data, request)
+        response = paginator.paginated_response
+
+        return response
+
+
+class SettingsTagMappingChildAddView(APIView):
+    permission_classes = (SettingsAccessPermission,)
+    serializer_class = TagMappingSerializer
+
+    @transaction.atomic
+    def put(self, request):
+        try:
+            parent_uuid = request.data.get("parent")
+            children_uuids = request.data.get("children", [])
+
+            parent = EnabledTagKeys.objects.get(uuid=parent_uuid, enabled=True)
+            children = EnabledTagKeys.objects.filter(uuid__in=children_uuids, enabled=True)
+
+            if not parent or not children.exists():
+                return Response({"detail": "Invalid parent or children UUIDs."}, status=status.HTTP_400_BAD_REQUEST)
+
+            # Check if any of the children_uuids are already in TagMapping
+            if TagMapping.objects.filter(child__uuid__in=children_uuids).exists():
+                return Response(
+                    {"detail": "Some child UUIDs are already inserted in TagMapping."},
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+
+            # Validate if parent can be a child in any existing mappings
+            if TagMapping.objects.filter(child=parent_uuid).exists():
+                return Response({"detail": "A child can't become a parent."}, status=status.HTTP_400_BAD_REQUEST)
+
+            # Validate if children can be parents in any existing mappings
+            if TagMapping.objects.filter(parent__in=children_uuids).exists():
+                return Response({"detail": "A parent can't become a child."}, status=status.HTTP_400_BAD_REQUEST)
+
+            with transaction.atomic():
+                # Create TagMapping for each child
+                for child in children:
+                    TagMapping.objects.create(parent=parent, child=child)
+
+            # Serialize and return the created TagMapping
+            serializer = TagMappingSerializer(TagMapping.objects.filter(parent=parent), many=True)
+
+            return Response(serializer.data, status=status.HTTP_203_NON_AUTHORITATIVE_INFORMATION)
+
+        except EnabledTagKeys.DoesNotExist:
+            return Response({"detail": "Invalid parent or children UUIDs."}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class SettingsTagMappingChildRemoveView(APIView):
+    permission_classes = (SettingsAccessPermission,)
+
+    def put(self, request: Request):
+        children_uuids = request.data.get("ids", [])
+        if not EnabledTagKeys.objects.filter(uuid__in=children_uuids).exists():
+            return Response({"detail": "Invalid children UUIDs."}, status=status.HTTP_400_BAD_REQUEST)
+
+        TagMapping.objects.filter(child__in=children_uuids).delete()
+
+        return Response(
+            {"detail": "Children deleted successfully."}, status=status.HTTP_203_NON_AUTHORITATIVE_INFORMATION
```

Comment:
This should probably be a Response(status=status.HTTP_204_NO_CONTENT). Does the 203 response really provide any additional information? We don't use this status code for any other PUT request throughout our repo.

Reply:
This is actually my fault @maskarb I told @bacciotti  the wrong return code, I thought 203 was no content and didn't double check myself before sending the slack. 😅 

Thanks for catching this @maskarb 🙇 
Sorry for leading you astray @bacciotti.